File: LevelSDKExampleApp/app/src/main/java/com/theshopatvsp/levelandroidsdk/ble/model/response/TransmitControlData.java
package com.theshopatvsp.levelandroidsdk.ble.model.response;

import  com.theshopatvsp.levelandroidsdk.ble.model.response.*;

/**
 * Created by andrco on 10/2/15.
 */
public class TransmitControlData extends DataPacket {
    private int totalByteCount;
    private int totalRecordCount;
    private byte writeData;

    public TransmitControlData(int totalRecordCount, int totalByteCount) {
        super();
        this.totalByteCount = totalByteCount;
        this.totalRecordCount = totalRecordCount;
    }

    public TransmitControlData(byte writeData) {
        super();
        this.writeData = writeData;
    }

    public int getTotalByteCount() {
        return totalByteCount;
    }

    public void setTotalByteCount(int totalByteCount) {
        this.totalByteCount = totalByteCount;
    }

    public int getTotalRecordCount() {
        return totalRecordCount;
    }

    public void setTotalRecordCount(int totalRecordCount) {
        this.totalRecordCount = totalRecordCount;
    }

    public byte[] getPacket() {
        byte bytes[] = new byte[1];

        bytes[0] = this.writeData;

        return bytes;
    }
}

File: LevelIOSSdkExample/LevelIOSSdkExample/LevelIOSSdk/model/TransmitControlData.swift
//
//  TransmitControlData.swift
//  LevelIOSSdkExample
//
//  Created by Andrew Cook on 11/1/16.
//  Copyright © 2016 TheShop. All rights reserved.
//

import Foundation

class TransmitControlData: DataPacket {
    var totalByteCount: Int
    var totalRecordCount: Int
    var writeData: UInt8?
    
    init(totalRecordCount: Int, totalByteCount: Int) {
        self.totalRecordCount = totalRecordCount
        self.totalByteCount = totalByteCount
        
        super.init()
    }
    
    convenience init(writeData: UInt8) {
        self.init(totalRecordCount: -1, totalByteCount: -1)
        
        self.writeData = writeData
    }
    
    override func getPacket() -> [UInt8] {
        var bytes: [UInt8] = [UInt8]()
        
        bytes.append(self.writeData!)
        
        return bytes
    }
}
